### PR TITLE
Add missing attributes of type

### PR DESCRIPTION
**Pull request by davidhalter**
All these attributes can be seen when using `dir(type)`.

In the future we should be discussing if certain methods on object (like
__eq__) should really be there. IMO this should be defined on type where it
actually also appears when using `dir`.

### CODE REVIEW

File: stdlib/3/builtins.pyi [hunk 1]
```diff
@@ -96,6 +103,10 @@ class type:
     def mro(self) -> List[type]: ...
     def __instancecheck__(self, instance: Any) -> bool: ...
     def __subclasscheck__(self, subclass: type) -> bool: ...
+    # there is a better way to specify the return type. It must
+    # be subscriptable, but it may not be a Mapping/MutableMapping.
```
Comment (JelleZijlstra) [inline stdlib/3/builtins.pyi]: Why can't it be a `Mapping`? It actually returns a `dict`, which is a Mapping.

Comment (JelleZijlstra) [reply stdlib/3/builtins.pyi]: I guess I read that as "it is not allowed to be a Mapping", which is not what you meant. :)

If we need to be really precise, we could use a Protocol (I suppose it needs to return something that implements `__getitem__`, `__setitem__`, and `__delitem__`.) But I think `MutableMapping` is fine.

**Comment by davidhalter** [comment]
Sorry for the huge delay. I have changed the `__prepare__` now. Not sure if I really need to deal with that comment. Let me know what you think.

File: stdlib/3/builtins.pyi (hunk 1 above)
Comment (davidhalter) [reply stdlib/3/builtins.pyi]: I have used Mapping now - I think that's more precise. It doesn't need to be mutable. Let me know if that's ok.

File: stdlib/3/builtins.pyi [hunk 2]
```diff
@@ -96,6 +103,8 @@ class type:
     def mro(self) -> List[type]: ...
     def __instancecheck__(self, instance: Any) -> bool: ...
     def __subclasscheck__(self, subclass: type) -> bool: ...
+    @classmethod
+    def __prepare__(metacls, __name: str, __bases: Tuple[type, ...], **kwds) -> Mapping[str, Any]: ...
```
Comment (srittau) [inline stdlib/3/builtins.pyi]: Add `Any` as annotation of `kwds`:

```suggestion
    def __prepare__(metacls, __name: str, __bases: Tuple[type, ...], **kwds: Any) -> Mapping[str, Any]: ...
```

**Pull request merged**